lib/micro-enum-DOCUMENTED.js

```diff
@@ -73,21 +73,15 @@
  * @version 1.0.0
  */
 
-var Enum = // the name of the "class"
-	function() // this is what the arguments are passed to
-	{
-		var values = arguments; // get the varargs and save them to a 'values' variable.
-		var self = { // prepare a 'self' object to return, so we work with an object instead of a function
-			all : [],     // prepare a list of all indices
-			keys : values // create the list of all keys
-		};
-		
-		for(var i = 0; i < values.length; i++) // for all enum names given
-		{
-			self[values[i]] = i; // add the variable to this object
-			self.all[i] = i;     // add the index to the list of all indices
-		}
-		
-		return self; // return the 'self' object, instead of this function
+const Enum = (...elements) => {
+	const s = {
+		all: [],
+		keys: elements
 	}
-;
\ No newline at end of file
+
+	elements.map((el, i) => {
+		s[el] = s.all[i] = i
+	})
+
+	return s
+}
```

Comment:
Thanks for modernizing this! The changes make a lot of sense!

However, this "DOCUMENTED" file is overly-compacted. It was originally unrolled and commented to help explain how the code works, so removing the comments, shortening variable names (`s`, `el`), and combining assignments (`s[el] = s.all[i] = i`) makes it more difficult to understand what's going on